File: C++/404. Sum of Left Leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
// Time: O(n)
class Solution {
public:
    int slove(TreeNode* node, int flag)
    {
        if(node==nullptr)
            return 0;
        if(node->left==nullptr && node->right==nullptr)
        {
            if(flag) return node->val;
            return 0;
        }
        return slove(node->right, 0)+slove(node->left, 1);
    }
    int sumOfLeftLeaves(TreeNode* root) {
        return slove(root, 0);
    }
};

File: Python/404. Sum of Left Leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
# Time: O(n)
class Solution:
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def solve(node, flag):
            if node is None:
                return 0
            if node.left is None and node.right is None:
                if flag == True:
                    return node.val
                return 0
            return solve(node.left, True) + solve(node.right, False)
        return solve(root, False)